avro4s-macros/src/main/scala/com/sksamuel/avro4s/FromRecord.scala

```diff
@@ -45,7 +45,11 @@ object FromValue extends LowPriorityFromValue {
       val decimalConversion = new Conversions.DecimalConversion
       val decimalType = LogicalTypes.decimal(sp.precision, sp.scale)
       override def apply(value: Any, field: Field): BigDecimal = {
-        decimalConversion.fromBytes(value.asInstanceOf[ByteBuffer], null, decimalType)
+        if (value == null && field.defaultVal() != null) {
+          decimalConversion.fromBytes(ByteBuffer.wrap(field.defaultVal.asInstanceOf[Array[Byte]]), null, decimalType)
+        } else {
+          decimalConversion.fromBytes(value.asInstanceOf[ByteBuffer], null, decimalType)
+        }
```

Comment:
Do you think the default value should be used on a read, or just a write?

Reply:
I think it should be for reading as well. I mean, if the value doesn't exist in the record, and there's a default in the schema, it may make sense to get the default.